source/common/http/conn_manager_impl.cc

```diff
@@ -917,7 +917,7 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
       // For Egress (outbound) response, if a decorator operation name has been provided, it
       // should be used to override the active span's operation.
       if (resp_operation_override) {
-        if (!resp_operation_override->value().empty()) {
+        if (!resp_operation_override->value().empty() && active_span_) {
```

Comment:
I think to hit this I think you'd have to encodeHeaders before calling decodeHeaders.

This likely happens during early response errors, dealing with bad responses.

I'd suggest reverting this change in your client, turning on egress tracing by default in the http integration tests (extending the default config in testing/config/utility.cc), and seeing if any tests fail.  If they do you can then reintroduce your change with confidence it is regression tested.

Reply:
cc @objectiser who wrote this code originally with the override. @mandarjog he might be able to help out with a test.